_overviews/overview/scala-tools.md

```diff
@@ -0,0 +1,320 @@
+
+>To install sbt, see [its download page](https://www.scala-sbt.org/download.html).
+
+
+### The sbt directory structure
+
+Like Maven, sbt uses a standard project directory structure. A nice benefit of that is that once you’re comfortable with its structure, it makes it easy to work on other Scala/sbt projects.
+
+The first thing to know is that underneath the root directory of your project, sbt expects a directory structure that looks like this:
+
+```bash
+build.sbt
+project/
+src/
+-- main/
+   |-- resources/
+   |-- scala/
+|-- test/
+   |-- resources/
+   |-- scala/
+target/
+```
+
+The *resources* directories are optional, and you can also add *java* directories at the same level as the *scala* directories if you want to use Java source code in your project. You can also add a *lib* directory under the root directory if you want to add unmanaged dependencies — JAR files — to your project.
+
+
+### Creating a “Hello, world” sbt project directory structure
+
+<!-- 
+  TODO: using something like `sbt new lampepfl/dotty.g8` may eventually
+        be preferable, but that seems to have a few bugs atm (creates
+        a 'target' directory above the root; renames the root dir;
+        uses 'dottyVersion'; 'name' doesn’t match the supplied name;
+        config syntax is a little hard for beginners.)
+-->
+
+Creating this directory structure is simple. There are tools to do this for you, but assuming that you’re using a Unix/Linux system, you can use these commands to create your first sbt project directory structure:
+
+```bash
+mkdir HelloWorld
+cd HelloWorld
+mkdir -p src/{main,test}/{resources,scala}
+mkdir project target
+```
+
+When you run a `find .` command after running those commands, you should see this result:
+
+```bash
+$ find .
+.
+./project
+./src
+./src/main
+./src/main/resources
+./src/main/scala
+./src/test
+./src/test/resources
+./src/test/scala
+./target
+```
+
+If you see that, you’re in great shape for the next step.
+
+>There are other ways to create the files and directories for an sbt project. One way is to use the `sbt new` command, [which is documented here on scala-sbt.org](https://www.scala-sbt.org/1.x/docs/Hello.html). That approach isn’t shown here because some of the files it creates are more complicated than necessary for an introduction like this.
+
+
+### Creating a first build.sbt file
+
+At this point you only need two more things to run a “Hello, world” project:
+
+- A *build.sbt* file
+- A *Hello.scala* file
+
+For a little project like this, the *build.sbt* file only needs to contain a few lines, like this:
+
+```scala
+name := "HelloWorld"
+version := "1.0"
+scalaVersion := "{{ site.scala-version }}"
```

Comment:
actually, for starters you don't even need the directory structure _or_ an sbt file. :) 

For experimenting, it is often enough to just have a Scala file and a `build.sbt` file that lists the scalaVersion and the dependencies.

Reply:
Maybe you can show the directory structure later? For people coming from scripting languages this always feels like a lot of boilerplate.

Reply:
Done